awkward/array/jagged.py

```diff
@@ -1350,9 +1362,42 @@ def concatenate(isclassmethod, cls_or_self, arrays):
 
         return cls(starts, stops, content)
 
-    @classmethod
-    def stack(cls, first, *rest):    # each item in first followed by second, etc.
-        raise NotImplementedError
+    @awkward.util.bothmethod
+    def stack(isclassmethod, cls_or_self, arrays):    # each item in first followed by second, etc.
+
+        if isclassmethod: 
+            cls = cls_or_self
+            if not all(isinstance(x, JaggedArray) for x in arrays):
+                raise TypeError("cannot stack non-JaggedArrays with JaggedArray.stack")
+        else:
+            self = cls_or_self
+            cls = self.__class__
+            if not isinstance(self, JaggedArray) or not all(isinstance(x, JaggedArray) for x in arrays):
+                raise TypeError("cannot stack non-JaggedArrays with JaggedArray.stack")
+            arrays = (self,) + tuple(arrays)
+
+        if len(set([len(a) for a in arrays])) > 1:
+            raise ValueError("cannot stack JaggedArrays of different lengths")
+
+        np = cls.numpy
+
+        n = len(arrays)
+        n_content = sum([len(a.flatten()) for a in arrays])
```

Comment:
Maybe flatten them once so that you don't have to re-flatten them. Flattening is trivial if the arrays have nice `starts` and `stops`, but if they're multidimensional, out of order, or whatever, it's not necessarily a fast operation.